### ISSUE TITLE
Tests

### ISSUE THREAD

**Issue by hussam789** [fmrsabino/movies-db-android#1]
I read you blog [posts](https://medium.com/@fmrsabino/an-approach-to-mvp-mvi-rxjava-2-dagger-2-part-2-of-2-one-chain-of-events-33e171096436), and I like this implementation of MVI/MVP with ModelView (Android arch component) and RxJava and state reducers to manage state (inspired by Jake Wharton's Presentation). 
It would be nice to see also Unit Test sample as mentioned in [Managing State with RxJava at 46:55](https://youtu.be/0IKHxjkgop4?t=46m56s)
**Testing Presenters (business logic of the app):**
![image](https://user-images.githubusercontent.com/1441832/27529928-0f16105a-5a5f-11e7-9ed4-9e8b6e6db149.png)

**Unit Testing UI and how it renders different states:**
This can be useful to see how to app looks like and behaves given a certain _**State (UI model)**_ 
![image](https://user-images.githubusercontent.com/1441832/27530064-9f579530-5a5f-11e7-8e65-c8bace72435c.png)

**Comment by fmrsabino** [comment]
Yes that's also on the plans but since I've been working on and off on this project it might take some time. Previous commits had tests for the MVP architecture if you are interested on those.
